internal/k8s/controller.go

```diff
@@ -1861,6 +1878,34 @@ func (lbc *LoadBalancerController) updateVirtualServerRoutesStatusFromEvents() e
 	return nil
 }
 
+func (lbc *LoadBalancerController) updatePolicyStatus() error {
+	var allErrs []error
+	for _, obj := range lbc.policyLister.List() {
+		pol := obj.(*conf_v1.Policy)
+
+		err := validation.ValidatePolicy(pol, lbc.isNginxPlus, lbc.enablePreviewPolicies)
+		if err != nil {
+			msg := fmt.Sprintf("Policy %v/%v is invalid and was rejected: %v", pol.Namespace, pol.Name, err)
+			err = lbc.statusUpdater.UpdatePolicyStatus(pol, conf_v1.StateInvalid, "Rejected", msg)
+			if err != nil {
+				allErrs = append(allErrs, err)
+			}
+		}
```

Comment:
there is a bug here - the code for reporting "AddedOrUpdated" will always be executed, even if the the error is not nil.

Shall it be like below?

```
if err != nil {
  // policy is rejected
} else {
 // policy is added or updated
}
```